libs/barista-components/filter-field/src/filter-field-multi-select/filter-field-multi-select-trigger.ts

```diff
@@ -0,0 +1,758 @@
+import {
+  DOWN_ARROW,
+  ENTER,
+  ESCAPE,
+  SPACE,
+  TAB,
+  UP_ARROW,
+} from '@angular/cdk/keycodes';
+import {
+  Overlay,
+  OverlayConfig,
+  OverlayContainer,
+  OverlayRef,
+  PositionStrategy,
+  ViewportRuler,
+} from '@angular/cdk/overlay';
+import { Platform } from '@angular/cdk/platform';
+import { DOCUMENT } from '@angular/common';
+import {
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  Host,
+  Inject,
+  Input,
+  NgZone,
+  OnDestroy,
+  Optional,
+} from '@angular/core';
+import { ControlValueAccessor } from '@angular/forms';
+import {
+  calculateOptionHeight,
+  DtAutocompleteOrigin,
+  DtOptionConfiguration,
+  DT_AUTOCOMPLETE_VALUE_ACCESSOR,
+  DT_OPTION_CONFIG,
+} from '@dynatrace/barista-components/autocomplete';
+import {
+  DtFlexibleConnectedPositionStrategy,
+  DtOption,
+  DtOptionSelectionChange,
+  dtSetUiTestAttribute,
+  DtUiTestConfiguration,
+  DtViewportResizer,
+  DT_UI_TEST_CONFIG,
+  isDefined,
+  stringify,
+  _countGroupLabelsBeforeOption,
+  _getOptionScrollPosition,
+  _readKeyCode,
+} from '@dynatrace/barista-components/core';
+import { DtFormField } from '@dynatrace/barista-components/form-field';
+import {
+  defer,
+  EMPTY,
+  fromEvent,
+  merge,
+  Observable,
+  of as observableOf,
+  Subject,
+  Subscription,
+} from 'rxjs';
+import {
+  delay,
+  filter,
+  map,
+  startWith,
+  switchMap,
+  take,
+  takeUntil,
+  tap,
+} from 'rxjs/operators';
+import { DtFilterFieldMultiSelect } from './filter-field-multi-select';
+
+@Directive({
+  selector: `input[dtFilterFieldMultiSelect]`,
+  exportAs: 'dtFilterFieldMultiSelectTrigger',
+  host: {
+    class: 'dt-multiselect-trigger',
+    '[attr.autocomplete]': 'autocompleteAttribute',
+    '[attr.aria-activedescendant]': 'activeOption?.id',
+    '[attr.aria-expanded]': 'multiSelectDisabled ? null : panelOpen.toString()',
+    '[attr.aria-owns]':
+      '(multiSelectDisabled || !panelOpen) ? null : multiSelect?.id',
+    '(focusin)': '_handleFocus()',
+    '(blur)': '_handleBlur()',
+    '(input)': '_handleInput($event)',
+    '(keydown)': '_handleKeydown($event)',
+  },
+  providers: [DT_AUTOCOMPLETE_VALUE_ACCESSOR],
+})
+export class DtFilterFieldMultiSelectTrigger<T>
+  implements ControlValueAccessor, OnDestroy {
+  private _optionHeight: number;
+  private _maxPanelHeight: number;
+
+  private _overlayRef: OverlayRef | null;
+  private _componentDestroyed = false;
+  private _overlayAttached = false;
+
+  /** The filter-field multiSelect panel to be attached to this trigger. */
+  @Input('dtFilterFieldMultiSelect')
+  get multiSelect(): DtFilterFieldMultiSelect<T> {
+    return this._multiSelect;
+  }
+  set multiSelect(value: DtFilterFieldMultiSelect<T>) {
+    this._multiSelect = value;
+    this._detachOverlay();
+  }
+  private _multiSelect: DtFilterFieldMultiSelect<T>;
+
+  /** autocomplete` attribute to be set on the input element. */
+  @Input('autocomplete') autocompleteAttribute = 'off';
```

Comment:
Which autocomplete is this. I don't see why the multi select trigger should require an autocomplete. Maybe you can explain this a bit. 
Imho this trigger does not need a lot more functionality than the range-trigger.

Reply:
My mistake. I removed it